### PR TITLE
[google_maps_flutter_platform_interface] Platform interface changes for #3258

### PR DESCRIPTION
**Pull request by Rexios80**
See #3258

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:6]: Nit: Our style is to leave a blank line between package and non-package dependencies.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:71]: Nit: "Web only." should be its own paragraph (like the comparable comments you're adding above), per Effective Dart style.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
The `applyDiff` tests look good, thanks! There should be tests for equality (that changing just each of the new properties causes objects not to be equal), hash (that changing just each of the new properties changes the hash) and `diffFrom` as well though.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart
Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:71]: Done

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Looks like you accidentally merged into another version update on the last update.

**Review by stuartmorgan — APPROVED** [review]
LGTM once the version merge is fixed, thanks!

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md
Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Although it looks like the pubspec version change is missing?

Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Okay I think I fixed it

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart
Comment (ditman) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:7]: Removing `import 'ui.dart';` is great, thanks!

**Review by ditman — APPROVED** [review]
LGTM!

**Pull request merged**
